File: client/lib/definitions/category.ts
export type GeneralCategoryModel = {
  id: string;
  name: string;
  description: string;
  slug: string;
  image_url: string;
  published: boolean;
  remarkable: boolean;
  created_at: Date;
  updated_at: Date;
  products?: ProductCategory[];
};

export type ProductCategory = {
  id: string;
  name: string;
  image_url?: string;
  description: string;
  slug: string;
};

File: internal/dto/category.go
package dto

type AdminCategoryDetail struct {
	ID          string            `json:"id"`
	Name        string            `json:"name"`
	Description *string           `json:"description,omitempty"`
	Slug        string            `json:"slug"`
	Published   bool              `json:"published,omitempty"`
	CreatedAt   string            `json:"createdAt,omitempty"`
	UpdatedAt   string            `json:"updatedAt,omitempty"`
	ImageUrl    *string           `json:"imageUrl,omitempty"`
	Products    []ProductListItem `json:"products,omitempty"`
}

type CategoryDetail struct {
	ID          string           `json:"id"`
	Name        string           `json:"name"`
	Description *string          `json:"description,omitempty"`
	Slug        string           `json:"slug"`
	Published   bool             `json:"published,omitempty"`
	CreatedAt   string           `json:"createdAt,omitempty"`
	ImageUrl    *string          `json:"imageUrl,omitempty"`
	Products    []ProductSummary `json:"products"`
}

type GeneralCategory struct {
	ID   string `json:"id"`
	Name string `json:"name"`
}
